sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2202,4 +2202,12 @@ class DataFrameSuite extends QueryTest with SharedSparkSession {
           |*(1) Range (0, 10, step=1, splits=2)""".stripMargin))
     }
   }
+
+  test("SPARK-28916: subexrepssion elimination can cause 64kb code limit") {
```

Comment:
how long does this test run? We can write a unit test instead if the end-to-end test is too expensive to run.

Reply:
yes, it takes about 3 mins. I'll try and find a way to create a UT...